core/src/main/java/com/linecorp/armeria/common/stream/StreamWriter.java

```diff
@@ -96,7 +94,6 @@ default void write(Supplier<? extends T> o) {
      * @throws IllegalArgumentException if the publication of the specified object has been rejected
      * @see <a href="#reference-counted">Life cycle of reference-counted objects</a>
      */
-    @CheckReturnValue
```

Comment:
I think we can remove this annotation rather than suppressing where it's used. 🤔

Reply:
/cc @anuraaga who suggested this annotation first. :wink:

Reply:
I think it's much easier to have a bug because of not checking this return value, in almost all cases the user should specially handle a closed stream right?

For example, here it looks like we should actually be skipping the rest of the callback if the stream is already closed

https://github.com/line/armeria/blob/4f62fc03a092d23de7ef8b3d8687120b15a5558d/core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java#L332

Reply:
Ah, that makes sense. Let me fix that. 😄

Reply:
Found out that we can just call `res.write(ping);` there. 😆